### PR TITLE
Gem cleanups and specs

### PR DESCRIPTION
**Pull request by toolmantim**
As promised, here's a PR with some specs! :)

```
$ rake spec
.....

Finished in 0.0025 seconds (files took 0.17947 seconds to load)
5 examples, 0 failures
```

It also this adds a bunch of standard Ruby gem'ish things, the biggest being this:

```
$ rake -T
rake build            # Build emoji_regex-0.1.0.gem into the pkg directory
rake clean            # Remove any temporary products
rake clobber          # Remove any generated files
rake install          # Build and install emoji_regex-0.1.0.gem into system gems
rake install:local    # Build and install emoji_regex-0.1.0.gem into system gems without network access
rake regenerate       # Regenerate the lib/*.rb files
rake release[remote]  # Create tag v0.1.0 and build and push emoji_regex-0.1.0.gem to Rubygems
rake spec             # Run RSpec code examples
```

It also checks in the generated `lib/emoji_regex.rb` file, which I'm thinking might be a good idea for a few reasons:

* Someone can easily check GitHub to see the thing that's inside the Gem they're using
* The lib can easily be tested in a branch/PR, using the actual lib files that will used for release
* PRs will actually show how the regexp has changed alongside the node lib

Strangely, the regenerate script doesn't create the same regex from the `lib` file released to RubyGems? But I tested all this by using the lib from RubyGems instead.

### CODE REVIEW

File: emoji_regex.gemspec [hunk 1]
```diff
@@ -9,4 +9,9 @@ Gem::Specification.new do |s|
   s.email       = 'hello@jessicastokes.net'
   s.license     = 'MIT'
   s.files       = ['lib/emoji_regex.rb']
+  s.require_paths = ['lib']
```
Comment (ticky) [inline emoji_regex.gemspec]: I was under the impression `require_paths` was implicit?

File: script/regenerate [hunk 2]
```diff
@@ -11,14 +13,14 @@ const replaceEscapes = (regexFactory) => (
...
-
+ 
 const input = fs.readFileSync(path.join(__dirname, '../src/emoji_regex.rb'), 'utf8');
-
+ 
 const output = input
   .replace('/% emojiRegex %/', emojiRegex)
   .replace('/% emojiRegexText %/', emojiRegexText);
-
+
```
Comment (ticky) [inline script/regenerate]: What’s with the whitespace here?

**Comment by ticky** [comment]
> Strangely, the regenerate script doesn't create the same regex from the lib file released to RubyGems? But I tested all this by using the lib from RubyGems instead.

Yep, unfortunately it won’t until mathiasbynens/emoji-regex#24 is fixed!

**Review by ticky — APPROVED** [review]

File: emoji_regex.gemspec (hunk 1 above)
Comment (toolmantim) [reply emoji_regex.gemspec]: Ah, maybe it is!

File: script/regenerate (hunk 2 above)
Comment (toolmantim) [reply script/regenerate]: VSCode just being naughty I think

**Comment by toolmantim** [comment]
Is there a fixed fork/branch we can pin our package.json to?

**Comment by toolmantim** [comment]
I'd be good to add a spec for `::Text` too, but I wasn't sure what it was used for and couldn't find a Readme example.

**Comment by ticky** [comment]
Unfortunately, because the problem is that the dependencies were incorrectly installed when the published version of the package was built, no.

**Comment by ticky** [comment]
[`::Text` will also find characters which shouldn’t be emoji by default](https://github.com/mathiasbynens/emoji-regex/blob/3233d998ef33161dd43dbdefe340767b80ce4712/test/tests.js#L87-L112) which is useful if you want to be more greedy about it. I think this test is a good start, though!

**Comment by toolmantim** [comment]
I cleaned up those few things you asked about. And added a `::Text` spec too (even though I still don't really understand what it does 🙃)

**Comment by toolmantim** [comment]
I pinned to a fixed version of the npm lib, so everything here builds and tests as expected. When the upstream is fixed, we can remove that pinning and bump to the fixed version!

**Review by ticky — APPROVED** [review]
Going to make some minor tweaks when I merge, but thanks! This is great :)

**Comment by toolmantim** [comment]
👏🏼 Thanks!!

**Pull request merged**